### DISCUSSION TITLE
Importing in a typescript project

### DISCUSSION THREAD

**Discussion by ohnoimissed** [Annoraaq/grid-engine#312] [Q&A] [answered]
Hello! I'm trying to use this in my phaser project using typescript via an import statement and I get the following error:
`Could not find a declaration file for module '[...]/GridEngine.esm.min.js'. '[...]/GridEngine.esm.min.js' implicitly has an 'any' type. ts(7016)`

My import statement is:
`import GridEngine from '[...]/GridEngine.esm.min.js';`

I've tried both **GridEngine.esm.min.js** and **GridEngine.min.js**, as well as copying some of the various .d.ts files into the folder with it, with no luck.

I'm sure the solution is going to be obvious, but I'm new to all this typescript stuff and hit a roadblock. Can anyone help? :) Thanks in advance.

**Comment by Annoraaq** [comment, marked as answer]
Are you using yarn or npm? In that case I would recommend to add grid-engine to your dependencies and just import it.

In your terminal:
```
npm i --save grid-engine
```

Then you can import it like:
```
import { GridEngine } from "grid-engine";
```

**Reply by ohnoimissed** [reply]
Thank you, that solved my issue.

Unfortunately I haven't managed to successfully use the functionality yet, I'm a bit unclear on how to make collision layers/character layers work despite going over the docs several times. Is it okay for me to open a new discussion?

**Reply by Annoraaq** [reply]
Of course. Feel free to open a new one. It might help others, too.
